spring-integration-mongodb/src/main/java/org/springframework/integration/mongodb/store/AbstractConfigurableMongoDbMessageStore.java

```diff
@@ -145,7 +161,9 @@ public void afterPropertiesSet() {
 		}
 
 		this.messageBuilderFactory = IntegrationUtils.getMessageBuilderFactory(this.applicationContext);
+	}
 
+	protected void createIndexes() {
```

Comment:
This method has to be called from this `afterPropertiesSet()`. And thanks to polymorphism an overridden  method is going to be called. Therefore we don't need to check for `createIndex` in the target implementation of this class.